common/sagas/notifications.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+import { takeEvery, put } from 'redux-saga/effects';
+import {delay} from 'redux-saga'
+import {closeNotification} from 'actions/notifications'
+import type {ShowNotificationAction} from 'actions/notifications'
+
+function* handleNotification(action: ShowNotificationAction) {
+    const {duration} = action.payload
+    // show forever
+    if (duration === 0) {
+        return
+    }
+
+    // FIXME
+    yield delay(duration || 5000)
```

Comment:
If I'm understanding this, it stops code execution here for duration or 5 seconds and then closes the modal...

What is the best way of fixing this?